_posts/2022-10-09-decentralization-of-zk-rollups.md

```diff
@@ -0,0 +1,192 @@
+
+To note is also the fact that any individual can become a sequencer by posting the L1 transaction. This does have benefits in terms of censorship resistance (if another sequencer refuses to process my transaction, I can just submit my own transaction), but it does come with some drawbacks. One is that due to this uncoordinated nature, it may not be obvious as to which sequencer I should submit my transaction to, in order for it to be confirmed quicker, taking into consideration that larger batches are more likely to be confirmed by aggregators quicker, and that some aggregators may wait to pool more L2 transactions before posting to L1 in order to reduce the per-unit gas cost. Additionally, the action of sequencers posting the L2 transaction data on L1s could have opened up the possibility for an inter-sequencer MEV attack. An observer of the L1 mempool can observe L2 transactions as they come in, and if they identify a transaction that is of potential MEV value they could have submitted another transaction with the same transactions but now with the desired order / inclusion of transactions and a higher gas tip. This means that the attacker's transaction would have been accepted before the honest sequencers’, resulting in the sequencer losing out on the reward of L2 gas, and having to pay for L1 gas without any reimbursement. But, the team has implemented a ChainID function that is specific to each sequencer, and is included in the signed transaction, so that a user only ever authorizes one sequencer (and only them) to execute a transaction, thereby mitigating the chance of a sandwich attack by another sequencer. 
+
+Once the sequencer has posted the transaction information on L1, it has now become the new virtual future state. In order for this to be confirmed, a validity proof must be submitted on L1 by an aggregator. This soft-confirmation by the virtual state, allows there to be a higher TPS, and lower block interval time. Without this soft-confirmation block interval time would be lengthened to the time taken to generate a proof, which is around 5 minutes, and coordination between provers would constrain throughput as there is no ordering of transactions that is decided on. This is conducted on a “first come, first accepted” basis, in that the aggregator who submits the proof first will be rewarded by the L2 transaction fees. This race format is concerning in that it encourages centralization around the fastest aggregator, which is the one with the most compute resources. 
```

Comment:
Why does the "virtual future state" needs to be submitted first without the proof? Is this a specific design, and what is the rationale behind the design decision? Also, where does the 5-minute come from? Is that based on a specific design? What exactly is the coordination between provers? Why is the ordering not decided at this stage? And why does the aggregator who submits the proof first gets rewarded by the layer 2 fees? Statements in this paragraph needs some references and explanations on the design choices

Reply:
If this flow is specific to the design in a single project instead of a general pattern, it should be clarified at the beginning of this subsection. The beginning of the section refers to Polygon Hermez proposals and some other projects, and is mostly discussing general issues. Here, in this paragraph, the approach (and its problems) seems to be specific to one particular project and its design

Reply:
The paragraph makes several assumptions: (1) the proving time is slow (5-10 minutes) (2) off-chain coordination is bad and inefficient (3) there is no way to distribute the reward to multiple aggregators. These assumptions should be made explicit. Also, they are specific to a project's design choice and parameters. Some of them probably don't even hold (such as 2 and 3, and maybe (1) in the near future).